### PR TITLE
Dynamic shape support for builder collapse

### PR DESCRIPTION
**Pull request by ggalieroc**

### CODE REVIEW

File: ngraph/core/src/builder/reshape.cpp [hunk 1]
```diff
@@ -202,3 +202,52 @@ shared_ptr<Node> builder::opset1::squeeze(const Output<Node>& value, vector<size
...
+{
+    if (start_axis == end_axis)
+    {
+        return value.get_node_shared_ptr();
+    }
+
+    if (value.get_partial_shape().is_static())
+    {
+        auto shape = value.get_shape();
+        size_t collapsed_axis_size = accumulate(next(begin(shape), start_axis),
```
Comment (tomdol) [inline ngraph/core/src/builder/reshape.cpp:218]: I'd like to see a short comment explaining what gets multiplied here. Can you add a single sentence above this line?

**Review by tomdol — APPROVED** [review]
LGTM overall

**Review by blesniewski — APPROVED** [review]

**Review by tsocha — APPROVED** [review]

**Review by jdanieck — APPROVED** [review]

File: ngraph/test/backend/builder_reduce_ops_opset1.in.cpp [hunk 2]
```diff
@@ -81,3 +81,94 @@ NGRAPH_TEST(${BACKEND_NAME}, builder_opset1_mean_dynamic_2)
...
+        const auto elems_in_tensor = shape_size(shape_input);
+
+        std::vector<float> input_values(elems_in_tensor, 1);
+        std::vector<float> expected_values(elems_in_tensor, 1);
+
+        test_case.add_input<float>(shape_input, {input_values});
+        test_case.add_expected_output<float>(shape_output, expected_values);
+        test_case.run();
+    }
+}
```
Comment (sdurawa) [inline ngraph/test/backend/builder_reduce_ops_opset1.in.cpp:174]: missing newline mark at the end of file

**Review by ilyachur — APPROVED** [review]

File: ngraph/test/backend/builder_reduce_ops_opset1.in.cpp (hunk 2 above)
Comment (ggalieroc) [reply ngraph/test/backend/builder_reduce_ops_opset1.in.cpp]: Done

File: ngraph/core/src/builder/reshape.cpp (hunk 1 above)
Comment (ggalieroc) [reply ngraph/core/src/builder/reshape.cpp:219]: Done

**Review by sdurawa — APPROVED** [review]

**Review by mitruska — APPROVED** [review]

**Review by postrational — APPROVED** [review]

**Pull request merged**